src/main/java/org/symphonyoss/symphony/messageml/MessageMLParser.java

```diff
@@ -470,15 +535,72 @@ public void loadElementId(String id) throws InvalidInputException {
   }
 
   /**
-   * Returns the label value corresponding to the id
+   * Returns the attributes corresponding to the id for a splittable element
    * (used internally during parsing)
    */
-  public Optional<String> getLabel(String id) throws InvalidInputException {
-    Optional<String> value = labels.get(id);
-    if(value == null){
-      throw new InvalidInputException(String.format("Label not found for the id: %s. The must be present before the corresponding element.", id));
+  public Optional<Map<String, String>> getSplittableAttributes(String id, Class<? extends SplittableElement> clazz) throws InvalidInputException {
+    SplittableData data = splittableComponents.get(id);
+    if(data == null){
+      return Optional.empty();
     }
-    return value;
+    return data.getAttributes(clazz);
+  }
+
+  /**
+   * Returns all attributes corresponding to the id
+   * (used internally during parsing)
+   */
+  public Optional<Map<Class<? extends SplittableElement>, Map<String, String>>> getAllSplittableAttributes(String id) throws InvalidInputException {
+    SplittableData data = splittableComponents.get(id);
+    if(data == null){
+      return Optional.empty();
+    }
+    return Optional.of(data.getAllAttributes());
+  }
+
+  /**
+   * Returns all values corresponding to the id
+   * (used internally during parsing)
+   */
+  public Optional<Map<Class<? extends SplittableElement>, Pair<String, String>>> getAllSplittableValues(String id) throws InvalidInputException {
+    SplittableData data = splittableComponents.get(id);
+    if(data == null){
+      return Optional.empty();
+    }
+    return Optional.of(data.getAllValues());
+  }
+
+  private void createSpittable(String id, Class<? extends SplittableElement> clazz, Pair<String,String> value)
```

Comment:
Typo in the method's name.

Reply:
Hi @lmacedodaitan I fixed this...and BTW another small bug, missing data-generated="true" attribute when converting to ProsentationML